spec/OpenLineage.json

```diff
@@ -64,6 +64,72 @@
         "schemaURL"
       ]
     },
+    "DatasetEvent": {
+      "type": "object",
+      "properties": {
+        "dataset": {
+          "allOf": [
+            { "$ref": "#/$defs/Dataset" }
```

Comment:
This is kind of workaround ;)

So, current code generation for Java works in a way such that base classes are generated as `interfaces` and only top level objects are classes. Dataset is only an interface while `InputDataset` and `OutputDataset` is a class. So, we cannot use `Dataset` directly here, instead we create an object with `allOf` containing `Dataset`. This advantage of this is that the existing code of generated client does not get modified. The disadvantage is a method name `newDatasetEventDataset` to create an instance of `DatasetEventDataset`  as in case of un-named objects, generated class name is a parent name concatenated with a property name.

Reply:
Or create a named `StaticDataset` type to use here as a ref

Reply:
I will try `StaticDataset` approach.